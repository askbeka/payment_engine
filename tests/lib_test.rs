use pe_lib::run;
use insta;
use itertools::Itertools;

#[test]
fn test_from_snapshots() {
    insta::glob!("csv_inputs/*.csv", |path| {
        let mut reader = csv::Reader::from_path(path).unwrap();
        let mut writer = csv::Writer::from_writer(vec![]);
        
        run(&mut reader, &mut writer).unwrap();

        let written = String::from_utf8(writer.into_inner().unwrap()).unwrap();
        let mut lines = written.lines();
        let csv = lines.next().unwrap();

        insta::assert_snapshot!(String::from(csv) + "\n" + &lines.sorted().join("\n"));
    });
}