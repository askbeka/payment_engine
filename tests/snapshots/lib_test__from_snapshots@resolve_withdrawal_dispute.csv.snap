---
source: tests/lib_test.rs
expression: "String::from(csv) + \"\\n\" + &lines.sorted().join(\"\\n\")"
input_file: tests/csv_inputs/resolve_withdrawal_dispute.csv
---
client,available,held,total,locked
1,0.5,0.0000,0.5,false
